admin_guide/backup_restore.adoc

```diff
@@ -0,0 +1,224 @@
+- etcd data on each master
+- API objects
+- registry storage
+- volume storage
+
+This topic does not cover how to back up and restore
+xref:../install_config/persistent_storage/index.adoc#install-config-persistent-storage-index[persistent
+storage], as those topics are left to the underlying storage provider.
+
+
+[[backup-restore-prerequisites]]
+== Prerequisites
+
+. Because the restore procedure involves a xref:cluster-restore[complete
+reinstallation], save all the files used in the initial installation. This may
+include:
++
+- *_~/.config/openshift/installer.cfg.yml_* (from the
+xref:../install_config/install/quick_install.adoc#install-config-install-quick-install[Quick Installation]
+method)
+- Ansible playbooks and inventory files (from the
+xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[Advanced
+Installation] method)
+- *_/etc/yum.repos.d/ose.repo_* (from the
+xref:../install_config/install/disconnected_install.adoc#install-config-install-disconnected-install[Disconnected
+Installation] method)
+
+. Install packages that provide various utility commands:
++
+----
+# yum install etcd
+----
+
+Note the location of the *etcd* data directory (or `$ETCD_DATA_DIR` in the
+following sections), which depends on how *etcd* is deployed.
+
+[options="header",cols="1,2"]
+|===
+| Deployment | Data Directory
+
+|all-in-one VM
+|*_/var/lib/openshift/openshift.local.etcd_*
+
+|external (not on master)
+|*_/var/lib/etcd_*
+
+|embedded (on master)
+|*_/var/lib/origin/etcd_*
+|===
+
+
+[[cluster-backup]]
+== Cluster Backup
+
+. Save all the certificates and keys, on each master:
++
+----
+# cd /etc/origin/master
+# tar cf /tmp/certs-and-keys-$(hostname).tar \
+    master.proxy-client.crt \
+    master.proxy-client.key \
+    proxyca.crt \
+    proxyca.key \
+    master.server.crt \
+    master.server.key \
+    ca.crt \
+    ca.key \
+    master.etcd-client.crt \
+    master.etcd-client.key \
+    master.etcd-ca.crt
+----
+
+. If *etcd* is running on more than one host, stop it on each host:
++
+----
+# sudo systemctl stop etcd
+----
++
+Although this step is not strictly necessary, doing so ensures that the *etcd*
+data is fully synchronized.
+
+. Create an *etcd* backup:
++
+----
+# etcdctl backup \
+    --data-dir $ETCD_DATA_DIR \
+    --backup-dir $ETCD_DATA_DIR.bak
+----
++
+[NOTE]
+====
+If *etcd* is running on more than one host,
+the various instances regularly synchronize their data,
+so creating a backup for one of them is sufficient.
+====
+
+. Create a template for all cluster API objects:
++
+====
+----
+$ oc export all \
+    --exact \//<1>
+    --all-namespaces \
+    --as-template=mycluster \//<2>
+    > mycluster.template.yaml
+----
+<1> Preserve fields that may be cluster specific,
+such as service `portalIP` values or generated names.
+<2> The output file has `kind: Template` and `metadata.name: mycluster`.
+====
++
+[IMPORTANT]
+====
+The object types included in `oc export all` are:
+
```

Comment:
@deads2k can you please take a look at this `oc export all` output? Is this correct? The formatting seems strange and I am not sure what was originally intended here. Thanks!